packages/cli/src/lib/version.ts

```diff
@@ -90,12 +90,6 @@ export function createPackageVersionProvider(lockfile?: Lockfile) {
     }
 
     const lockfileEntries = lockfile?.get(name);
-    if (
-      name.startsWith('@types/') &&
-      lockfileEntries?.some(entry => entry.range === '*')
-    ) {
-      return '*';
```

Comment:
This path is only being hit in one template right now (assuming it's not used anywhere else other than as a result of `versionQuery` expressions in templates). I'm proposing pulling out as a blunt first try to fix the e2e test issue.

Reply:
I'm thinking it might be best if we filter out * from the logic below that finds a range that satisfies the target version, and only use it if it is the only option?

need a changeset for this change too btw

Reply:
> I'm thinking it might be best if we filter out * from the logic below that finds a range that satisfies the target version, and only use it if it is the only option?

That's on top of this change, right? If so, should we perhaps keep that separate in the hope to keep this change somewhat simple? The change in dab696a already fixes the issue at hand in the code, and I'm running out of time to put into this today.

> need a changeset for this change too btw

Yep, ofc. Will do that once we're definite on the change being made.

Reply:
No the risk is introduced by the fact that we're no longer exiting on `*`. I think it's fine though because the logic is wrong in a way where `*` won't be prioritized as it picks the lowest compatible range instead of the highest.

:shipit: 🤷

Reply:
:+1: changeset added.